### PR TITLE
Fix issue #284

### PR DESCRIPTION
**Pull request by elfman**

### CODE REVIEW

**Comment by birdofpreyru** [comment]
@elfman Does not look working with MM for me. Probably, because MM challenges don't have registration dates, as the regular challenges do, so your updated code is just skipped for them, and they are still shown as stalled.

**Comment by elfman** [comment]
I am really confused with the challenge data, why challenge's `status` has been set to `PAST`, but the field `registrationOpen` is still true, what is true state of this challenge?
145aec03-c19b-4e3d-8342-d783d3af471a

**Comment by elfman** [comment]
In some regular challenges, the registration date has past, why the value of `registrationOpen` is still `Yes`. Which filed should I use?
691b8609-9b37-4615-ba2e-e6ce69206496

**Comment by birdofpreyru** [comment]
@elfman Please double-check the code related to fetching challenges. I believe, what happens is that `registrationOpen` field is not actually fetched from the backend: regular challenges and MMs are fetched from different api endpoints by challenge service, and then they are passed through a normalization function, which ensures that these objects have the same format (by adding missing fields, and doing any other modifications, necessary for the rest of the code to handle these objects in the same way). I guess, `registrationOpen` field is added there, and there might be an error with setting it for some challenges (probably, it is actually not used anymore around the code, then should be removed).

**Comment by elfman** [comment]
@birdofpreyru I checked the origin data before `normalizeChallenge`, and I don't understand why the `scheduledEndTime` of `Registration` is different from `registrationEndDate`
67b4518f-8d9a-4bb4-af62-566732427047

**Comment by birdofpreyru** [comment]
@elfman Please re-check with production backend. Dev backend has problems with sync between v2 and v3 APIs, probably this is the reason.

**Comment by elfman** [comment]
@birdofpreyru Updated, please check again.

File: src/shared/services/challenges.js [hunk 1]
```diff
@@ -79,7 +79,8 @@ export function normalizeMarathonMatch(challenge, username) {
     numSubmissions: challenge.userIds ? challenge.userIds.length : 0,
     platforms: '',
     prizes: [0],
-    registrationOpen: endTimestamp > Date.now() ? 'Yes' : 'No',
+    registrationOpen: endTimestamp > Date.now() &&
+      !(challenge.status && challenge.status === 'PAST') ? 'Yes' : 'No',
```
Comment (birdofpreyru) [inline src/shared/services/challenges.js]: @elfman `!(challenge.status && challenge.status === 'PAST')` is equivalent to just `challenges.status !== 'PAST'`, which is surely more clean, and thus preffered code. Will update myself, though.

**Pull request merged**